packages/genesys-spark-components/src/components/stable/gux-listbox/gux-listbox.service.ts

```diff
@@ -38,23 +54,53 @@ function setActiveOption(
   }
 }
 
+function getSiblingForGroupedList(
+  list: HTMLGuxListboxElement,
+  location: 'next' | 'previous' | 'first' | 'last'
+): ListboxOptionElement {
+  const availableOptionsList = getListOptions(list).filter(
+    option => !option.disabled && !option.filtered
+  );
+  const activeOption = getActiveOption(list);
+  const activeOptionIndex = availableOptionsList.indexOf(activeOption);
+
+  switch (location.toUpperCase()) {
+    case 'FIRST':
+      return availableOptionsList[0];
+    case 'LAST':
+      return availableOptionsList[availableOptionsList.length - 1];
+    case 'PREVIOUS':
+      return availableOptionsList[activeOptionIndex - 1];
+    default:
+      return availableOptionsList[activeOptionIndex + 1];
+  }
+}
+
 function getFirstOption(list: HTMLGuxListboxElement): ListboxOptionElement {
```

Comment:
I think we may be able to simplify these function. I will give you a call on zoom later to discuss.